BaragonKubernetesIntegration/src/main/java/com/hubspot/baragon/kubernetes/listeners/DeploymentAddedListener.java

```diff
@@ -0,0 +1,66 @@
+package com.hubspot.baragon.kubernetes.listeners;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Optional;
+import com.hubspot.baragon.models.BaragonRequest;
+import com.hubspot.baragon.models.BaragonService;
+import com.hubspot.baragon.models.UpstreamInfo;
+
+import io.fabric8.kubernetes.api.model.Event;
+import io.fabric8.kubernetes.api.model.OwnerReference;
+
+public class DeploymentAddedListener implements KubernetesEventListener {
+  private static final Logger LOG = LoggerFactory.getLogger(DeploymentAddedListener.class);
+
+  // TODO: Verify that these are the field names Orion is placing in K8s metadata
```

Comment:
remember Orion is HubSpot specific, these will need to be configurable. We can add a k8s section of the config yaml for Baragon with defaults of the values we will use. The config yaml will also likely need a enabled boolean since not everyone will be using this module